client/src/chat/chatApiClient.ts

```diff
@@ -35,26 +35,28 @@ class ChatApiClient extends ApiClientBase {
 		// I'm confused about how to use the WsMessageEvent
 		// For now I just hard code the message to be in the same form as WsMessageEvent
 
-		const data: WsMessageEvent = {
-			dataType: "message",
-			operation: "add",
-			context: id,
-			data: {
-				id: "123", // how to get this?
-				from: "username", // how to get this?
-				message: message,
-			},
-		};
+		// const data: WsMessageEvent = {
+		// 	dataType: "message",
+		// 	operation: "add",
+		// 	context: id,
+		// 	data: {
+		// 		id: "123", // how to get this?
+		// 		from: "username", // how to get this?
+		// 		message: message,
+		// 	},
+		// };
 
-		return (await this.wsSendMessage(data)) as Thread;
+		this.wsSendMessage({ threadId: id, message });
 	}
 }
 
 export function useChatApi() {
 	const [user] = useUser();
 	const [apiEndpoint] = useApiEndpoint();
 
-	const { sendJsonMessage } = useWebSocket(apiEndpoint.ws);
+	const { sendJsonMessage } = useWebSocket(apiEndpoint.ws, {
```

Comment:
I'm kinda doubting whether useWebSocket is the best api after all, because of how react will re-render stuff based on this updating. Maybe late to look into it now though